### PR TITLE
Add unit tests for AdoClient

### LINKED ISSUE #177: Unit Test AdoClient
We need to add unit tests for `AdoClient.cs`. 

### Acceptance Criteria
- [x] Every public method needs to be covered by a unit test.

### PR DESCRIPTION
**Pull request by ArinGhazarian**
Closes #177 

This PR adds unit test for `AdoClient`. 

- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked

### CODE REVIEW

File: src/OctoshiftCLI.Tests/AdoClientTests.cs [hunk 1]
```diff
@@ -0,0 +1,1042 @@
...
+            using var firstHttpResponse = new HttpResponseMessage(HttpStatusCode.OK)
+            {
+                Headers = { RetryAfter = new RetryConditionHeaderValue(TimeSpan.FromSeconds(1)) },
+                Content = new StringContent("FIRST_RESPONSE")
+            };
+            using var secondHttpResponse = new HttpResponseMessage(HttpStatusCode.OK)
+            {
+                Content = new StringContent("SECOND_RESPONSE")
+            };
+            using var thridResponse = new HttpResponseMessage(HttpStatusCode.OK)
```
Comment (iomekam) [inline src/OctoshiftCLI.Tests/AdoClientTests.cs:89]:
```suggestion
            using var thirdResponse = new HttpResponseMessage(HttpStatusCode.OK)
```

File: src/OctoshiftCLI.Tests/AdoClientTests.cs [hunk 2]
```diff
@@ -0,0 +1,1042 @@
...
```
Comment (iomekam) [inline src/OctoshiftCLI.Tests/AdoClientTests.cs:214]:
```suggestion
            using var thirdResponse = new HttpResponseMessage(HttpStatusCode.OK)
```

File: src/OctoshiftCLI.Tests/AdoClientTests.cs [hunk 3]
```diff
@@ -0,0 +1,1042 @@
...
+                .Protected()
+                .SetupSequence<Task<HttpResponseMessage>>(
+                    "SendAsync",
+                    ItExpr.Is<HttpRequestMessage>(req =>
+                        req.Method == HttpMethod.Post &&
+                        req.Content.ReadAsStringAsync().Result == EXPECTED_JSON_REQUEST_BODY),
+                    ItExpr.IsAny<CancellationToken>())
+                .ReturnsAsync(firstHttpResponse)
+                .ReturnsAsync(secondHttpResponse)
+                .ReturnsAsync(thridResponse);
```
Comment (iomekam) [inline src/OctoshiftCLI.Tests/AdoClientTests.cs:229]:
```suggestion
                .ReturnsAsync(thirdResponse);
```

File: src/OctoshiftCLI.Tests/AdoClientTests.cs [hunk 4]
```diff
@@ -0,0 +1,1042 @@
...
+            var handlerMock = new Mock<HttpMessageHandler>();
+            handlerMock
+                .Protected()
+                .SetupSequence<Task<HttpResponseMessage>>(
+                    "SendAsync",
+                    ItExpr.Is<HttpRequestMessage>(req => req.Method == HttpMethod.Get),
+                    ItExpr.IsAny<CancellationToken>())
+                .ReturnsAsync(firstHttpResponse)
+                .ReturnsAsync(secondHttpResponse)
+                .ReturnsAsync(thridResponse);
```
Comment (iomekam) [inline src/OctoshiftCLI.Tests/AdoClientTests.cs:102]:
```suggestion
                .ReturnsAsync(thirdResponse);
```

File: src/OctoshiftCLI.Tests/AdoClientTests.cs [hunk 5]
```diff
@@ -0,0 +1,1042 @@
...
+            _loggerMock.Verify(m => m.LogWarning("THROTTLING IN EFFECT. Waiting 1000 ms"), Times.Once);
+        }
+
+        [Fact]
+        public async Task GetAsync_Logs_The_Url()
+        {
+            // Arrange
+            using var httpClient = new HttpClient(MockHttpHandlerForGet().Object);
+            var adoClient = new AdoClient(_loggerMock.Object, httpClient, PERSONAL_ACCESS_TOKEN);
+            const string url = "http://example.com/resource";
```
Comment (iomekam) [inline src/OctoshiftCLI.Tests/AdoClientTests.cs:122]: I notice you use this URL pretty often. Is it worth making it a CONST?

File: src/OctoshiftCLI.Tests/AdoClientTests.cs [hunk 6]
```diff
@@ -0,0 +1,1042 @@
...
+            using var firstHttpResponse = new HttpResponseMessage(HttpStatusCode.OK)
+            {
+                Headers = { RetryAfter = new RetryConditionHeaderValue(TimeSpan.FromSeconds(1)) },
+                Content = new StringContent("FIRST_RESPONSE")
+            };
+            using var secondHttpResponse = new HttpResponseMessage(HttpStatusCode.OK)
+            {
+                Content = new StringContent("SECOND_RESPONSE")
+            };
+            using var thridResponse = new HttpResponseMessage(HttpStatusCode.OK)
```
Comment (iomekam) [inline src/OctoshiftCLI.Tests/AdoClientTests.cs:494]:
```suggestion
            using var thirdResponse = new HttpResponseMessage(HttpStatusCode.OK)
```

File: src/OctoshiftCLI.Tests/AdoClientTests.cs [hunk 7]
```diff
@@ -0,0 +1,1042 @@
...
+                .Protected()
+                .SetupSequence<Task<HttpResponseMessage>>(
+                    "SendAsync",
+                    ItExpr.Is<HttpRequestMessage>(req =>
+                        req.Method == HttpMethod.Patch &&
+                        req.Content.ReadAsStringAsync().Result == EXPECTED_JSON_REQUEST_BODY),
+                    ItExpr.IsAny<CancellationToken>())
+                .ReturnsAsync(firstHttpResponse)
+                .ReturnsAsync(secondHttpResponse)
+                .ReturnsAsync(thridResponse);
```
Comment (iomekam) [inline src/OctoshiftCLI.Tests/AdoClientTests.cs:509]:
```suggestion
                .ReturnsAsync(thirdResponse);
```

File: src/OctoshiftCLI.Tests/AdoClientTests.cs (hunk 6 above)
Comment (iomekam) [reply src/OctoshiftCLI.Tests/AdoClientTests.cs:494]: I'll stop here. Looks like this mistake is for each HTTP method (PUT, POST, GET, etc)

**Review by iomekam — APPROVED** [review]
LGTM.  Left some minor comments

File: src/OctoshiftCLI.Tests/AdoClientTests.cs (hunk 1 above)
Comment (ArinGhazarian) [reply src/OctoshiftCLI.Tests/AdoClientTests.cs:89]: Oh thanks for catching, will fix 😃

**Review by dylan-smith — APPROVED** [review]
I'm gonna be honest, I skimmed through the unit tests pretty quick, you know what you're doing when it comes to testing, and anything is gonna be an improvement over the zero we had before this.

**Pull request merged**